hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHANodeDetails.java

```diff
@@ -144,7 +151,8 @@ public static SCMHANodeDetails loadDefaultConfig(
     return new SCMHANodeDetails(scmNodeDetails, Collections.emptyList());
   }
 
-  public static SCMHANodeDetails loadSCMHAConfig(OzoneConfiguration conf)
+  public static SCMHANodeDetails loadSCMHAConfig(
+      OzoneConfiguration conf, Optional<SCMStorageConfig> storageConfig)
```

Comment:
Looks like the Optional is only empty for tests right? Can we make `storageConfig` a required parameter and have the tests pass `new SCMStorageConfig(conf)`? This makes it clear to the production code that they must specify a storage config for correct behavior.